src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -0,0 +1,687 @@
+
+	volatile             int                                        acquired;
+	@SuppressWarnings("rawtypes")
+	private static final AtomicIntegerFieldUpdater<SimpleDequePool> ACQUIRED =
+			AtomicIntegerFieldUpdater.newUpdater(SimpleDequePool.class, "acquired");
+
+	volatile             int                                        wip;
+	@SuppressWarnings("rawtypes")
+	private static final AtomicIntegerFieldUpdater<SimpleDequePool> WIP =
+			AtomicIntegerFieldUpdater.newUpdater(SimpleDequePool.class, "wip");
+
+	volatile             ConcurrentLinkedDeque<Borrower<POOLABLE>> pending;
+	@SuppressWarnings("rawtypes")
+	private static final AtomicReferenceFieldUpdater<SimpleDequePool, ConcurrentLinkedDeque> PENDING =
+			AtomicReferenceFieldUpdater.newUpdater(SimpleDequePool.class, ConcurrentLinkedDeque.class, "pending");
+
+	SimpleDequePool(PoolConfig<POOLABLE> poolConfig, boolean pendingBorrowerFirstInFirstServed) {
+		super(poolConfig, Loggers.getLogger(SimpleDequePool.class));
+		this.idleResourceLeastRecentlyUsed = poolConfig.reuseIdleResourcesInLruOrder();
+		this.pendingBorrowerFirstInFirstServed = pendingBorrowerFirstInFirstServed;
+		this.pending = new ConcurrentLinkedDeque<>(); //unbounded
+		this.idleResources = new ConcurrentLinkedDeque<>();
+	}
+
+	@Override
+	public Mono<PooledRef<POOLABLE>> acquire() {
+		return new QueueBorrowerMono<>(this,
+				Duration.ZERO); //the mono is unknown to the pool until requested
+	}
+
+	@Override
+	public Mono<PooledRef<POOLABLE>> acquire(Duration timeout) {
+		return new QueueBorrowerMono<>(this,
+				timeout); //the mono is unknown to the pool until requested
+	}
+
+	@Override
+	public int acquiredSize() {
+		return acquired;
+	}
+
+	@Override
+	public Mono<Void> disposeLater() {
+		return Mono.defer(() -> {
+			@SuppressWarnings("unchecked") ConcurrentLinkedDeque<Borrower<POOLABLE>> q =
+					PENDING.getAndSet(this, TERMINATED);
+			if (q != TERMINATED) {
+				Borrower<POOLABLE> p;
+				while ((p = q.pollFirst()) != null) {
+					p.fail(new PoolShutdownException());
+				}
+
+				@SuppressWarnings("unchecked")
+				Queue<QueuePooledRef<POOLABLE>> e =
+						IDLE_RESOURCES.getAndSet(this, null);
+				if (e != null) {
+					Mono<Void> destroyMonos = Mono.empty();
+					while (!e.isEmpty()) {
+						QueuePooledRef<POOLABLE> ref = e.poll();
+						if (ref.markInvalidate()) {
+							destroyMonos = destroyMonos.and(destroyPoolable(ref));
+						}
+					}
+					return destroyMonos;
+				}
+			}
+			return Mono.empty();
+		});
+	}
+
+	@Override
+	public int idleSize() {
+		Queue<?> e = IDLE_RESOURCES.get(this);
+		return e == null ? 0 : e.size();
+	}
+
+	public void evictMatching(Predicate<PooledRef<POOLABLE>> predicate) {
+		Iterator<QueuePooledRef<POOLABLE>> iterator = idleResources.descendingIterator();
+		int count = 0;
+		while (iterator.hasNext()) {
+			QueuePooledRef<POOLABLE> next =  iterator.next();
+			if (predicate.test(next)) {
+				System.out.println("background eviction of " + next);
```

Comment:
May be replace with `log.debug`

Reply:
oh snap that whole method is a leftover of an experiment 😨